superset/result_set.py

```diff
@@ -174,7 +174,13 @@ def convert_pa_dtype(pa_dtype: pa.DataType) -> Optional[str]:
 
     @staticmethod
     def convert_table_to_df(table: pa.Table) -> pd.DataFrame:
-        return table.to_pandas(integer_object_nulls=True)
+        try:
+            return table.to_pandas(integer_object_nulls=True)
+        except pa.lib.ArrowInvalid:
+            return table.to_pandas(
+                integer_object_nulls=True,
+                timestamp_as_object=True
```

Comment:
Should we also consider `date_as_object`? https://arrow.apache.org/docs/python/generated/pyarrow.Table.html#pyarrow.Table.to_pandas

Reply:
It is true already by default:
![image](https://user-images.githubusercontent.com/30076090/155333731-63f00165-8097-4a0e-8518-cd00ea587aaf.png)

Reply:
Oh, thanks for pointing that out, I didn't notice it was already enabled by default! 🤦